src/components/userComponents/Footer/Footer.tsx

```diff
@@ -0,0 +1,557 @@
+              </svg>
+            </Link>
+          </div>
+        </div>
+      </div>
+      <div className="flex mt-12">
+        <p className="text-silver ml-56 text-base font-normal font-['Lato']">
+          Copyright © 2024 Peninsula Humane Society & SPCA{' '}
+        </p>
+        <Link
+          href="https://phs-spca.org/privacy-policy/"
+          className="ml-8 pb-10 text-center flex justify-center text-silver text-normal font-normal font-['Lato'] underline"
+        >
+          Privacy Policy
+        </Link>
+      </div>
+    </div>
+  );
+}
+/**
+ * @param root0
+ * @param root0.backLink
+ * @param root0
+ * @param root0.backLink
+ * @returns an email pop up.
+ * if no email is entered and the user clicks the submit button, an error message will pop up.
+ * if an invalid email is entered and the user clicks the submit button, another error message will pop up.
+ * otherwise, if a valid email is submitted and properly subscribed, another pop up will appear that will tell the user they are subscribed and direct them to another page.
+ */
+export default function Footer() {
+  const isWebDevice = useWebDeviceDetection();
+  const [inputValueName, setNameValue] = useState('');
+  const [inputValueEmail, setEmailValue] = useState('');
+  const [showError, setShowError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+  const [subscribed, setSubscribed] = useState(false);
+
+  const handleNameChange = e => {
+    setNameValue(e.target.value);
+    setShowError(false);
+  };
+
+  const handleEmailChange = e => {
+    setEmailValue(e.target.value);
+    setShowError(false);
+  };
+
+  // const { error } = await supabase.from('emails').insert({ id: 1, name: 'Denmark' })
+
+  const isValidEmail = email => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.(com|ca)$/;
+    return emailRegex.test(email);
+  };
+
+  const handleSubmit = async e => {
+    e.preventDefault();
+    // check name
+    if (!inputValueName.trim()) {
+      setShowError(true);
+      document.getElementById('nameInput').style.borderColor = 'red';
+      setErrorMsg('Please enter a name');
+    } else if (!inputValueEmail.trim()) {
+      // Handle the case for an empty email
+      setShowError(true);
+      document.getElementById('emailInput').style.borderColor = 'red';
+      setErrorMsg('Please enter an email address');
+    } else if (isValidEmail(inputValueEmail)) {
+      // Handle the submission for a valid email
+      try {
+        // Insert the email into the Supabase database
+        const { error } = await supabase
+          .from('emails')
+          .insert({ emails: inputValueEmail, first_name: inputValueName });
+        console.log('successfully inserted?');
+      } catch (error) {
+        console.error(error);
+        return error;
```

Comment:
@jxmoose Lets aim to shorten length of this file.  If file stills seems very large I would recommend splitting this component into a helper file or two for readability